CMakeLists.txt

```diff
@@ -195,72 +195,60 @@ target_link_libraries(
         numa
 )
 
-# Note on flags:
-#   ttnn and tt_lib will break if built with LTO, so leaving -fno-lto in compile options
-add_library(linker_flags INTERFACE)
-
-add_library(compiler_warnings INTERFACE)
-target_compile_options(
-    compiler_warnings
-    INTERFACE
-        -Werror
-        -Wdelete-non-virtual-dtor
-        -Wreturn-type
-        -Wswitch
-        -Wuninitialized
-        -Wno-unused-parameter
-)
-
-# add additional compile warning flags depending on the compiler
-ADJUST_METAL_COMPILER_WARNINGS()
-
-add_library(compiler_flags INTERFACE)
-target_link_libraries(
-    compiler_flags
-    INTERFACE
-        linker_flags
-        compiler_warnings
+string(TOUPPER "$ENV{ARCH_NAME}" ARCH_NAME_DEF)
+add_compile_definitions(ARCH_${ARCH_NAME_DEF})
+add_compile_options(
+    -Werror
+    -Wdelete-non-virtual-dtor
+    -Wreturn-type
+    -Wswitch
+    -Wuninitialized
+    -Wno-unused-parameter
+    -mavx2
+    -fPIC
+    -fvisibility-inlines-hidden
+    -fno-lto # FIXME: This seems to be here for ttnn; it should go to TTNN, then.
```

Comment:
To take a look in future. LTO is usually really good. Would be good to enable this.

Reply:
Agreed!  (it won't do any favours to our link times, though...)

But that's a separate effort.